src/ComboBox/ComboBox.svelte

```diff
@@ -299,11 +300,33 @@
                 selectedId = filteredItems[highlightedIndex].id;
               }
             } else {
-              open = false;
-              if (filteredItems[0]) {
-                value = itemToString(filteredItems[0]);
-                selectedItem = filteredItems[0];
-                selectedId = filteredItems[0].id;
+              // searching typed value in text list with lowercase
+              /** @type Array<ComboBoxItem> */
+              const listOfItems = filteredItems.map((e) => {
+                return {
+                  id: e.id,
+                  text: e.text?.toLowerCase(),
+                  disabled: e.disabled,
+                };
+              });
+              const matchedItem = listOfItems.find(
+                (e) => e.text === value?.toLowerCase()
+              );
+              if (matchedItem) {
+                // typed value has matched
+                open = false;
+                selectedItem = filteredItems.find(
+                  (e) => e.id === matchedItem.id
+                );
```

Comment:
`listOfItems` seems unnecessary.
- `e.text?.toLowerCase()` could be done directly in a `find` on filteredItems
- The additional `filteredItems.find` would not be needed and `matchedItem` could be assigned directly to `selectedItem`
- Also, the `disabled` is disregarded when determining matching items. Those probably should not be considered potential matches?

Reply:
Nice catch – items that are `disabled` should not be matched.

Reply:
Very good comments, thank you!